### PR TITLE
fix: repair auto-discovered sidebar agents

### PR DESCRIPTION
**Pull request by EtanHey**
## Summary
- Teach registry repair to convert no-suffix auto-discovered panes using parsed CLI plus title-derived repo evidence, including leading-marker titles like `🤝 driverBuddy`.
- Prevent duplicate live launcher-title surfaces from fighting over one canonical seat record by creating surface-scoped repaired registrations when the canonical seat is already live elsewhere.
- Downgrade `orchestrator_not_leftmost` and `worker_in_leftmost_column` to info severity for auto-discovered agents, while preserving blocking topology health for managed agents.

## Test plan
- [x] Red-first focused tests for no-suffix repair, duplicate-title collision, and auto-discovered topology severity.
- [x] `bun test tests/agent-registry.test.ts tests/agent-health.test.ts`
- [x] `bun test tests/resync-tool.test.ts tests/surface-topology.test.ts`
- [x] `bun run test && bun run typecheck && bun run build` — 82 test files / 1502 tests passed, then typecheck and build exited 0.
- [x] Pre-push hook reran tests — 82 test files / 1502 tests passed.
- [x] `coderabbit review --agent` — findings: 0.

## Notes
- Tool names stay unchanged; no pane moves.
- Full daemon convergence/live fleet resync remains outside this PR per the build brief.

---

> [!NOTE]
> **Medium Risk**
> Changes agent registry merge/repair and eviction paths that determine which agents appear and which surfaces are suppressed; behavior is covered by new tests but mistakes could mis-bind seats or drop rows.
> 
> **Overview**
> Fixes **auto-discovered sidebar agents** that never matched managed seat repair because pane titles lacked launcher suffixes (e.g. `🤝 driverBuddy`). Repair now falls back to **parsed CLI plus a repo token from the title** when the usual title→launcher inference fails.
> 
> **Registry reconciliation** gains seat/launcher **identity keys** and uses them to collapse duplicate managed rows, evict pending ghosts superseded by a real seat, and **skip creating a second registration** when the canonical seat is already bound to a different live surface. **`listMerged`** additionally **self-heals** eligible managed records onto a replacement surface when the old one closes, suppresses extra auto-rows for duplicate launcher titles, and syncs lifecycle state after repair.
> 
> **Agent health** treats **`orchestrator_not_leftmost`** and **`worker_in_leftmost_column`** as **info** (not blocking) for auto-discovered agents so sidebar placement noise stays healthy while managed agents keep strict topology checks.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 9e7e0ecb7098564fba9f564cae248b36ee55304b. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

> [!NOTE]
> ### Fix auto-discovered sidebar agent repair with identity-based self-healing and deduplication
> - Downgrades `orchestrator_not_leftmost` and `worker_in_leftmost_column` issue severities to `info` for auto-discovered agents in [agent-health.ts](https://github.com/EtanHey/cmuxlayer/pull/268/files#diff-0dd564610d06e111ffe28c5699d77a1cc6e6b41bdbe8d893f63099d89f87e8f3), so column-placement issues no longer degrade their health status.
> - Adds identity key matching (by seat ID and launcher name) in [agent-registry.ts](https://github.com/EtanHey/cmuxlayer/pull/268/files#diff-0d60b018bf3077cf672b99b42a9e57bddafe6ceb878621ab79b5ce52ab374aa6) so managed records can be re-associated to a new live surface when their original surface closes (self-healing).
> - Suppresses duplicate auto records when a discovered surface matches an existing managed identity, and evicts duplicate managed registrations keeping a single canonical record.
> - Extends launcher inference to handle surfaces without explicit suffixes by combining a repo token derived from the surface title with a CLI-specific suffix.
> - Risk: Self-healing rewrites managed record state and lifecycle in place; records in `done` state (except `error` with a `Surface ` prefix) are ineligible and will not be reassigned.
>
> 
>
> <sup><a href="https://app.macroscope.com">Macroscope</a> summarized 9e7e0ec.</sup>
> 
>

## Summary by CodeRabbit

* **New Features**
  * Auto-discovered agents now show reduced-severity health feedback for specific column-placement/liveness issues (while remaining healthy).

* **Bug Fixes**
  * Registry repair/self-heal is more robust: improved launcher inference, identity-based deduplication, conflict suppression across live surfaces, and safer lifecycle reset for repaired records.
  * Resync behavior for title-to-repo mapping edge cases is clearer, reducing confusing state mismatches.

* **Tests**
  * Expanded coverage for the new health severity rules and expanded repair/self-heal scenarios, plus updated resync fixtures.

### CODE REVIEW

**Comment by EtanHey** [comment]
@coderabbitai review

**Comment by EtanHey** [comment]
@cursor @bugbot review

**Comment by EtanHey** [comment]
B3 update pushed in 66b9a1a.

What changed:
- Added TDD coverage for the M1 mimir specimen: two auto Claude rows plus three stale pending ghosts collapse to exactly `mimirClaude` + `mimirCodex`.
- Added one-row-per-seat repair behavior: duplicate discovered surfaces for the same launcher/seat are suppressed instead of becoming `*-surface-*` rows.
- GC now evicts stale pending ghosts once their launcher/seat is represented by a managed row.

Verification:
- `bun test tests/agent-registry.test.ts` passed.
- `bun run test && bun run typecheck && bun run build` passed locally: 82 files / 1503 tests.
- Pre-push hook reran tests and passed: 82 files / 1503 tests.
- `coderabbit review --agent` completed with 0 findings.

**Comment by EtanHey** [comment]
Blocking review fix pushed in 9e7e0ec.

What changed:
- Removed the persisted `repairSuppressedSurfaceRefs` state.
- `listMerged` now derives duplicate suppression from the current discovery snapshot.
- If the canonical managed surface A is gone and duplicate live surface B remains for the same launcher/seat identity, `listMerged` self-heals the managed row onto B without requiring manual resync.
- Added regression coverage for A canonical -> A closes -> B visible via plain `listMerged`.

Verification:
- Red test failed before the fix with `cmuxlayerLead` still on closed `surface:35` and `Surface surface:35 disappeared`.
- `bun test tests/agent-registry.test.ts` passed: 26 tests.
- Related cluster passed: `bun test tests/agent-registry.test.ts tests/resync-tool.test.ts tests/agent-health.test.ts tests/surface-topology.test.ts` -> 72 tests.
- `bun run test && bun run typecheck && bun run build` passed locally: 82 files / 1504 tests.
- `coderabbit review --agent`: 0 findings.
- Pre-push hook reran tests and passed: 82 files / 1504 tests.

File: src/agent-registry.ts
[the remark this answers is not in the document]
Comment (EtanHey) [reply src/agent-registry.ts:1]: Fixed in 9e7e0ec. The persisted repairSuppressedSurfaceRefs field was removed; listMerged now derives duplicate suppression from the current discovery snapshot and self-heals the managed row onto the remaining live duplicate when the prior canonical surface disappears.

File: src/agent-registry.ts [hunk 1]
```diff
@@ -516,6 +570,9 @@ export class AgentRegistry {
       ) {
         continue;
       }
+      if (this.repairSuppressedSurfaceRefs.has(discoveredEntry.surface_id)) {
+        continue;
```
[the remark this answers is not in the document]
Comment (EtanHey) [reply src/agent-registry.ts]: Fixed in 9e7e0ec. Suppression is no longer persisted across scans; it is recomputed per listMerged call from live discovery evidence, with regression coverage for A canonical -> A closes -> B visible without repair.
